os/drivers/ai-soc/ndp120/src/ndp120_api.c

```diff
@@ -1231,19 +1231,25 @@ int ndp120_kd_start_match_process(struct ndp120_dev_s *dev) {
     return s;
 }
 
-int ndp120_start_sample_ready(struct ndp120_dev_s *dev)
+int ndp120_start_sample_ready(struct ndp120_dev_s *dev, bool resume)
 {
 	int s;
 
 	dev->recording = true;
 
+	/* reset the count of semaphore used to signal sample ready */	
+	sem_init(&dev->sample_ready_signal, 1, 0);
+
 	s =  ndp120_set_sample_ready_int(dev, 1);
 
 	uint32_t bytes_before_match = 0;
 
-	s = syntiant_ndp_extract_data(dev->ndp, SYNTIANT_NDP_EXTRACT_TYPE_INPUT,
+	if (!resume) {
```

Comment:
``
ndp120_start_sample_ready(priv);
```
priv->running = true;
```
and then
```
if (!dev->running)
```

Reply:
I have updated it.